sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/IndentingXMLStreamWriter.scala

```diff
@@ -0,0 +1,208 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```

Comment:
Is this your original work? I'm wondering if this code is `Apache License 2.0` or not, @aldenlau-db .

Reply:
Hi, this code is partially based on the original IndentingXMLStreamWriter provided here with logic translated to Scala: https://storage.googleapis.com/oss-fuzz-coverage/jaxb/reports/20240220/linux/com.sun.xml.txw2.output/IndentingXMLStreamWriter.java.html .

Reply:
Hi, I just talked to @sandip-db to clarify usage of logic from the original package--I'm going to close this PR and open a new one with new logic not based on `com.sun.xml.txw2.output.IndentingXMLStreamWriter` as I didn't realize it couldn't be used due to its license, sorry for the misunderstanding!

Reply:
Thank you, @aldenlau-db !